blade-graphics/src/vulkan/init.rs

```diff
@@ -341,14 +341,13 @@ impl super::Context {
             entry.create_instance(&create_info, None).unwrap()
         };
 
-        let is_xorg = match surface_handles {
-            Some((_, raw_window_handle::RawDisplayHandle::Xlib(_))) => true,
-            Some((_, raw_window_handle::RawDisplayHandle::Xcb(_))) => true,
-            _ => false,
-        };
         let bugs = SystemBugs {
-            intel_unable_to_present_on_xorg: is_xorg && is_nvidia_prime_forced(),
+            //Note: this is somewhat broad across X11/Wayland and different drivers.
+            // It could be narrower, but at the end of the day if the user forced Prime
+            // for GLX it should be safe to assume they want it for Vulkan as well.
+            intel_unable_to_present: is_nvidia_prime_forced(),
```

Comment:
This only applies to Xorg. Wayland completely ignores this, but using it as an indication of desired behaviour may work here.

Reply:
The general expected behaviour of desktop applications is that they default to igpu unless one of
```
PrefersNonDefaultGPU=false
X-KDE-RunOnDiscreteGpu=false
```
is set in the application.desktop file, or some desktops provide a method to offload with right-click menu.

However when it comes to games, typically vulkan defaults to dgpu. While opengl requires intervention.